app/components/ContactForm.js

```diff
@@ -0,0 +1,185 @@
+"use client";
+import PropTypes from "prop-types";
+import { useState } from "react";
+import { useRouter } from "next/navigation";
+import { ContactAPI } from "../data/contactAPI";
+
+/**
+ * This form is used in two scenarios: one for creating a new contact and adding it to the ContactAPI, and one for editing an existing contact
+
+So as not to copy/paste the same form in two places, this form has checks to see if the contact already exists in the ContactAPI object, and behaves differently based on whether or not that is true
+
+The form will be empty (all inputs set to null) if the contact does not exist, or populated with the existing contact information if the contact does exist
+ * 
+ * @component
+ * @param {Object} props - the component accepts contact and removeContact as props
+ * @param {Object} props.contact - an object that contains the contact's id, name, email, and phone number
+ * @returns {JSX.Element} the rendered form, either empty or populated with existing contact information
+ * 
+**/
+
+export default function ContactForm({
+  contact = { id: 0, name: "", email: "", phone_number: "", image_url: "" },
+}) {
+  const router = useRouter();
+  // set initial state of input values based on whether contact exists in ContactAPI
+  const initialName = ContactAPI.contacts.includes(contact)
+    ? contact.name
+    : null;
+  const initialEmail = ContactAPI.contacts.includes(contact)
+    ? contact.email
+    : null;
+  const initialPhone = ContactAPI.contacts.includes(contact)
+    ? contact.phone_number
+    : null;
+  const initialImgUrl = ContactAPI.contacts.includes(contact)
+    ? contact.image_url
+    : null;
+  const [name, setName] = useState(initialName);
+  const [email, setEmail] = useState(initialEmail);
+  const [phone_number, setPhone_number] = useState(initialPhone);
+  const [image_url, setImage_url] = useState(initialImgUrl);
+  const generateId = () => Math.round(Math.random() * 100000000);
```

Comment:
this should be in a utils separate file and not here because will recreate every time you access the page